pkg/ingester/errors.go

```diff
@@ -525,6 +527,19 @@ func (e ingesterTooBusyError) errorCause() mimirpb.ErrorCause {
 // Ensure that ingesterTooBusyError is an ingesterError.
 var _ ingesterError = ingesterTooBusyError{}
 
+type ingesterPushAPIDisabledError struct{}
+
+func (e ingesterPushAPIDisabledError) Error() string {
+	return ingesterPushAPIDisabledMsg
+}
+
+func (e ingesterPushAPIDisabledError) errorCause() mimirpb.ErrorCause {
+	return mimirpb.SERVICE_UNAVAILABLE
```

Comment:
`SERVICE_UNAVAILABLE` is treated as a server-side error, so retryable. I'm wondering if we should add a `NOT_ALLOWED` error, and then we map it as `405` status code and treat it as non retryable.